### PR TITLE
New feature: Rename pde to fit with Matlab DG-SparseGrid

### LINKED ISSUE #344: renamed PDEs
sync up fokkerplanck PDEs in C++ with renames to the MATLAB versions.

`fokkerplanck1_4p1a -> fokkerplanck1_pitch_E(1)` ,
`fokkerplanck1_4p1b -> fokkerplanck1_pitch_E(2)` , 
`fokkerplanck1_4p2-> fokkerplanck1_pitch_C`

### PR DESCRIPTION
**Pull request by hbrunie**
## Proposed changes

This PR is here to solve [the rename pde issue#344](https://github.com/project-asgard/asgard/issues/344).
It involves adding a parameter to asgard binary.

```
  --case_selection <e.g. 1 (or 2)>      PDE case to solve; default is 1,
                                            sometimes there is a second case
                                            possible.
```

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [x] New feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [x] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

**Comment by hbrunie** [comment]
I have rebased this PR above the branch of @bmcdanie fixing the time in analytic solution generation.
Once @bmcdanie PR would have been merged to asgard/develop I will have to rebase on develop and force push again.

**Comment by hbrunie** [comment]
To pass the tests `coefficients` and `time_advance`, we need the gold generated corresponding files for `fokkerplanck1_pitch_E_case2` (gaussian) from the Matlab `DG-SparseGrid` repository.

This is not implemented yet. This is the corresponding [PR] (https://github.com/project-asgard/DG-SparseGrid/pull/65) from this other repo.

File: src/program_options.hpp [hunk 1]
```diff
@@ -250,6 +267,7 @@ class parser
   std::string pde_str = DEFAULT_PDE_STR;
   // pde to construct/evaluate
   PDE_opts pde_choice = DEFAULT_PDE_OPT;
+  // pde selected case (f0)
```
Comment (bmcdanie) [inline src/program_options.hpp]: what does this comment refer to?

File: src/program_options.hpp [hunk 2]
```diff
@@ -111,18 +127,19 @@ class parser
...
+  static auto constexpr DEFAULT_TIME_STEPS        = 10;
+  static auto constexpr DEFAULT_WRITE_FREQ        = 0;
+  static auto constexpr DEFAULT_USE_IMPLICIT      = false;
+  static auto constexpr DEFAULT_USE_FG            = false;
+  static auto constexpr DEFAULT_DO_POISSON        = false;
+  static auto constexpr DEFAULT_DO_ADAPT          = false;
+  static auto constexpr DEFAULT_PDE_STR           = "continuity_2";
+  static auto constexpr DEFAULT_PDE_OPT           = PDE_opts::continuity_2;
+  static auto constexpr DEFAULT_SOLVER            = solve_opts::direct;
+  static auto constexpr DEFAULT_PDE_SELECTED_CASE = 0;
```
Comment (bmcdanie) [inline src/program_options.hpp]: can you init this with the enum value instead of an integer?

File: src/tensors.hpp [hunk 3]
```diff
@@ -150,13 +150,13 @@ class vector
...
   // instead of disabling w/ sfinae for const_view,
-  // static tools::expect added to definition
+  // static expect added to definition
   vector<P, mem, resrc> &operator=(vector<P, mem, resrc> const &);
 
   // move constructor/assignment (required to be same to same)
   vector(vector<P, mem, resrc> &&);
 
-  // as with copy assignment, static tools::expect added
+  // as with copy assignment, static expect added
```
Comment (bmcdanie) [inline src/tensors.hpp]: I believe this is my mistake originally - should say "static assert" but my sed changes to "tools::expect", then your sed changed that to "static expect".

File: src/tensors.hpp [hunk 4]
```diff
@@ -150,13 +150,13 @@ class vector
   vector(vector<P, mem, resrc> const &);
   // cannot be templated per C++ spec 12.8
   // instead of disabling w/ sfinae for const_view,
-  // static tools::expect added to definition
+  // static expect added to definition
```
Comment (bmcdanie) [inline src/tensors.hpp]: "static expect" -> "static assert"

File: src/pde/pde_fokkerplanck1_pitch_E.hpp [hunk 5]
```diff
@@ -68,19 +68,24 @@ class PDE_fokkerplanck_1d_4p1a : public PDE<P>
...
-      auto p  = phi(z(i), t);
-      auto t1 = 1 - std::pow(p, 2);
-      auto t2 = 1 - std::pow(z(i), 2);
-      auto t3 = f0(p);
-      f(i)    = t1 / t2 * t3;
+      P p  = phi(z(i), t);
+      P t1 = 1 - std::pow(p, 2);
+      P t2 = 1 - std::pow(z(i), 2);
+      P t3 = 0.;
+      if constexpr (user_case == PDE_case_opts::mod1)
```
Comment (bmcdanie) [inline src/pde/pde_fokkerplanck1_pitch_E.hpp]: we have a convention that we always use curlies around conditionals, even single line ones

**Review by bmcdanie — CHANGES_REQUESTED** [review]
thanks for this PR! couple of minor notes, looks great overall.

File: src/program_options.hpp (hunk 1 above)
Comment (hbrunie) [reply src/program_options.hpp]: I forgot to remove ..

File: src/program_options.hpp (hunk 2 above)
Comment (hbrunie) [reply src/program_options.hpp]: Yes.

File: src/tensors.hpp (hunk 3 above)
Comment (hbrunie) [reply src/tensors.hpp]: OK.

Comment (hbrunie) [reply src/tensors.hpp]: I also changed the one line 153.

File: src/tensors.hpp (hunk 4 above)
Comment (hbrunie) [reply src/tensors.hpp]: Yes

File: src/pde/pde_fokkerplanck1_pitch_E.hpp (hunk 5 above)
Comment (hbrunie) [reply src/pde/pde_fokkerplanck1_pitch_E.hpp]: yes

**Pull request merged**